docs: add documentation for github rate limits (#78) (#217)
liatrio/liatrio-otel-collector: pkg/receiver/gitproviderreceiver/README.md

|----------|-------------------------|
| [github] | Git Metrics from [GitHub](https://github.com/) |
| [gitlab] | Git Metrics from [GitLab](https://gitlab.com) |

## Rate Limiting

> Given this receiver scrapes data from Git providers it is subject to rate

Reviewer: ```suggestion
Given this receiver scrapes data from Git providers, it is subject to rate
```

> limiting. The following sections will give some sensible defaults for each

Reviewer: ```suggestion
limiting. The following section will give some sensible defaults for each
```

Is it sections? I only see a github one

git provider.

### GitHub

> This receiver primarily interacts with GitHub's GraphQL API. The default rate

Reviewer: ```suggestion
The GitHub scraper within this receiver primarily interacts with GitHub's GraphQL API. The default rate
```

limit for GraphQL API is 5,000 points per hour (unless your PAT is associated
to a GitHub Enterprise Cloud organization then your limit is 10,000).
The receiver on average costs 4 points per repository. This means that the
receiver can at most scrape 1250 repositories per hour.

Given this average cost a good collection interval in seconds is

```go

> (repository_count * 4 / hourly_graphql_rate / 3600) + 300

Reviewer: ```suggestion
(repository_count * 4 / hourly_graphql_rate / 3600) + 300
```

Thinking we can turn lines 134 -> 136 into a math expression. Maybe something like 

$` (n*4/r/3600)+300 `$

Where: 
n = number of repositories
r = hourly rate (5,000 or 10,000)

```

`hourly_graphql_rate` is likely 5000 but there are factors that can change this,
for more information see [GitHub's docs](https://docs.github.com/en/graphql/overview/rate-limits-and-node-limits-for-the-graphql-api#primary-rate-limit).
The `300` is a buffer to account for this being a rough estimate and to account
for the initial query to grab repositories.

> It is recommended to use the `search_query` config option to limit the number of

Reviewer: Love this portion. Very clear, succinct, and descriptive.

> repositories that are scraped. We recommend one receiver per team (note: `team`

Reviewer: ```suggestion
repositories that are scraped. We recommend one instantiation of the receiver per team (note: `team`
```

is not a valid quantifier when searching repositories `topic` is). Reminder that

> each receiver should have its own `GITHUB_PAT` as this is what rate limits are

Reviewer: ```suggestion
each instance of the receiver should have its own corresponding token for authentication as this is what rate limits are
```
